src/java.base/share/classes/java/security/DigestInputStream.java

```diff
@@ -51,6 +51,15 @@
  * retain a handle onto the digest object, and clone it for each
  * digest to be computed, leaving the original digest untouched.
  *
+ * @implNote This implementation updates and only updates the message digest
```

Comment:
My feeling still remains.  I don't recall seeing a wording construction like this except for IFF, which is well-understood.  I think it could be said more compactly with:

`This implementation only updates the message digest with data actually read from the input stream when it is on. `

Just my $0.02.

Reply:
For your consideration, something like:
 "... This implementation will update the message digest if and only if the stream is on ...."